### PR TITLE
--lf now skips colletion of files without failed tests

### LINKED ISSUE #5172: last-failed: smarter collection
With `--lf` it should not run a full collection, but use the information from the cache.

### PR DESCRIPTION
**Pull request by nicoddemus**
Fix #5172

### CODE REVIEW

File: src/_pytest/cacheprovider.py [hunk 1]
```diff
@@ -158,6 +158,33 @@ def __init__(self, config):
...
+        Ignore this file path if we are in --lf mode and it is not in the list of
+        previously failed files.
+        """
+        if not self.active or not self.config.getoption("lf") or not path.isfile():
+            return
+        last_failed_paths = self.last_failed_paths()
+        if not last_failed_paths:
+            # if no previous failures, we want to run all items regardless
+            return
+        result = Path(path) not in self.last_failed_paths()
```
Comment (blueyed) [inline src/_pytest/cacheprovider.py]:
```suggestion
        result = Path(path) not in last_failed_paths
```

Comment (nicoddemus) [reply src/_pytest/cacheprovider.py]: Good catch, thanks; done slightly different, I check for `self.lastfailed` on the `if`

**Comment by blueyed** [comment]
Thanks again.. this will help me with working on the pytest plugin / conftest to use pytest with Django's own tests (> 6000 IIRC), and basically all unittest based.
Ref: https://code.djangoproject.com/ticket/30415

**Comment by nicoddemus** [comment]
Awesome!

**Comment by blueyed** [comment]
I think this might have caused https://github.com/pytest-dev/pytest/issues/5301 - started looking into it, but might not finish it today anymore.

**Pull request merged**